ModularTegustation/tegu_mobs/apocalypse_bird.dm

```diff
@@ -499,33 +514,101 @@
 		if(M.z == z && M.client)
 			flash_color(M, flash_color = "#CCBBBB", flash_time = 50)
 	addtimer(CALLBACK(GLOBAL_PROC, .proc/show_global_blurb, 5 SECONDS, "A long time ago, in a warm and dense forest lived three happy birds.", 25))
+	COOLDOWN_START(src, speak_bird, 10 SECONDS)
+	COOLDOWN_START(src, summon_bird, 30 SECONDS)
+	force_bird_time = world.time + 3 MINUTES
+
+/mob/living/simple_animal/forest_portal/Life()
+	. = ..()
+	var/list/range_area = orange(2, src)
+	if(force_bird_time < world.time)
+		range_area = GLOB.abnormality_mob_list // If they take too long, we shove them in MANUALLY.
+	for(var/mob/living/simple_animal/hostile/abnormality/A in range_area) // We're just gonna suckem in...
+		if(A in stored_birds["unspoken"])
+			continue
+		if(A in stored_birds["spoken"])
+			continue
+		if(A.type in bird_types)
+			if(!ConsumeBird(A))
+				continue
+	if(COOLDOWN_FINISHED(src, speak_bird))
+		SpeakBird()
+	if(COOLDOWN_FINISHED(src, summon_bird))
+		SummonBird()
+		COOLDOWN_START(src, summon_bird, 30 SECONDS) // So they keep trying to move towards the portal, even if temporarily blocked.
 
 /mob/living/simple_animal/forest_portal/Bumped(atom/movable/AM)
 	if(!isliving(AM))
 		return ..()
+	if(!ishostile(AM))
```

Comment:
Any reasons for this?

Reply:
Thought it may reduce error cases of the ConsumeBird proc trying to use it on non-hostiles and forces normal interaction from non-hostiles guaranteed